### DISCUSSION TITLE
不能控制 formily/antd 中 upload 组件的上传按钮

### DISCUSSION THREAD

**Discussion by autumnLeaves0** [alibaba/formily#2347] [Q&A 2.x] [answered]
![image](https://user-images.githubusercontent.com/26240134/138673124-dd04f32c-000e-4859-8773-91dd3b620f8a.png)
antd的upload 可以控制不显示上传按钮，formily/antd 包装后就不能控制了

**Comment by janryWang** [comment, marked as answer]
你children传个空div就行了

**Reply by autumnLeaves0** [reply]
空div不行，antd会有元素包裹空div。
必须传null或undefined，但是传null或undefinedh会被 formily/antd/upload 改成 `getPlaceholder(props)`

**Reply by janryWang** [reply]
什么场景需要隐藏按钮呀？啥都隐藏了用户怎么点击

**Reply by autumnLeaves0** [reply]
当上传照片数到达限制后，上传按钮消失。
![image](https://user-images.githubusercontent.com/26240134/138990991-b3a17639-9e8b-47f0-9590-0ceaafa60d98.png)

**Reply by janryWang** [reply]
这个逻辑不是Antd内部自己消化的？？

**Reply by autumnLeaves0** [reply]
`formily/antd/upload` 包装了 `antd/upload` , 传null或undefined会被 formily/antd/upload 改成 `getPlaceholder(props)`，导致无法隐藏按钮。
![image](https://user-images.githubusercontent.com/26240134/138994217-21c53fc5-7f14-4028-b051-62a6a542a7ab.png)

**Comment by TheKenS** [comment]
老哥，请问你解决了吗？
